### PR TITLE
feat: l1ToL2MessageNonce

### LINKED ISSUE #98: feat: `l1ToL2MessageNonce`
# Feature Request

**Describe the Feature Request**

As a user, I would like to query `l1ToL2MessageNonce` of the `StarkNetCore` contract.

**Describe Preferred Solution**

`l1ToL2MessageNonce` is accessible in the core library, and as a CLI command, and a REST API endpoint.

**Describe Alternatives**

**Related Code**

- `beerus_core`
- `beerus_cli`
- `beerus_rest_api`

- [X] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by ClementWalter**
# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing
- [ ] Other (please describe):

# What is the current behavior?

Feature not implemented

Resolves #98 

# What is the new behavior?

Implemented in core, cli & api

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

# Other information

I was not sure about the key of the response JSON in the API : used `result`, should it be `nonce` ?

### CODE REVIEW

File: beerus_cli/src/starknet/mod.rs [hunk 1]
```diff
@@ -146,6 +146,21 @@ pub async fn query_starknet_l1_to_l2_messages(
     ))
 }
 
+/// Query l1 to l2 message nonce
+/// # Arguments
+/// * `beerus` - The Beerus light client.
+/// # Returns
+/// * `Result<()>` - The result of the query.
```
Comment (LucasLvy) [inline beerus_cli/src/starknet/mod.rs:153]:
```suggestion
/// * `Result<CommandResponse>` - The result of the query.
```

**Review by LucasLvy — CHANGES_REQUESTED** [review]

File: beerus_cli/tests/cli.rs [hunk 2]
```diff
@@ -736,6 +736,45 @@ mod test {
         assert_eq!("Chain id: 123", result.to_string());
     }
 
+    /// Test the `starknet_l1_to_l2_message_nonce` CLI command.
+    /// Given normal conditions, when query nonce, then ok.
+    /// Success case.
+    #[tokio::test]
```
Comment (abdelhamidbakhta) [inline beerus_cli/tests/cli.rs:742]: Please add error cases

File: beerus_cli/src/model.rs [hunk 3]
```diff
@@ -108,6 +108,8 @@ pub enum StarkNetSubCommands {
         #[arg(short, long, value_name = "MSG_HASH")]
         msg_hash: String,
     },
+    L1ToL2MessageNonce {},
+    /// The nonce of the L1 to L2 message bridge
```
Comment (abdelhamidbakhta) [inline beerus_cli/src/model.rs:112]: Comment should be above the line

File: beerus_core/tests/beerus.rs [hunk 4]
```diff
@@ -996,6 +996,40 @@ mod tests {
         assert_eq!(result.unwrap_err().to_string(), expected_error);
     }
 
+    /// Test the `starknet_l1_to_l2_message_nonce` method when everything is fine.
+    /// This test mocks external dependencies.
+    #[tokio::test]
+    async fn given_normal_conditions_when_call_get_l1_to_l2_message_nonce_then_should_return_ok() {
```
Comment (abdelhamidbakhta) [inline beerus_core/tests/beerus.rs:1002]: Please add error cases

File: beerus_rest_api/tests/api.rs [hunk 5]
```diff
@@ -887,6 +887,51 @@ mod test {
         );
     }
 
+    /// Test the `query_l1_to_l2_message_nonce` endpoint.
+    /// `/starknet/messaging/l1_to_l2_message_nonce`
+    /// Given normal conditions, when query_l1_to_l2_messages, then ok.
+    #[tokio::test]
```
Comment (abdelhamidbakhta) [inline beerus_rest_api/tests/api.rs:893]: Please add error cases

**Review by abdelhamidbakhta — CHANGES_REQUESTED** [review]
Please address comments

**Comment by ClementWalter** [comment]
I'm doing the error cases now; I left them apart first because I was not sure that these my ongoing work were correctly written, so I wanted a first approval to avoid to much rework, thxs!

**Comment by abdelhamidbakhta** [comment]
> I'm doing the error cases now; I left them apart first because I was not sure that these my ongoing work were correctly written, so I wanted a first approval to avoid to much rework, thxs!

Makes sense, thanks

**Review by abdelhamidbakhta — APPROVED** [review]
lgtm
please resolve conflicts

**Review by abdelhamidbakhta — APPROVED** [review]
lgtm

**Review by abdelhamidbakhta — APPROVED** [review]
lgtm

**Pull request merged**